doc/subsystems/shell/shell.rst

```diff
@@ -239,127 +239,23 @@ Simple command handler implementation:
 	}
 
 .. warning::
-	Do not use function :cpp:func:`shell_fprintf` outside of the command
-	handler because this might lead to incorrect text display on the
-	screen. If any text should be displayed outside of the command context,
-	then use the :ref:`logger`.
+	Do not use function :cpp:func:`shell_fprintf` or shell print macros:
+	:c:macro:`shell_print`, :c:macro:`shell_info`, :c:macro:`shell_warn`,
+	:c:macro:`shell_error` outside of the command handler because this
+	might lead to incorrect text display on the screen. If any text should
+	be displayed outside of the command context, then use the
+	:ref:`logger` or `printk` function.
 
 Command help
 ------------
 
-Every user-defined command, subcommand, or option can have its own help
-description. The help for commands and subcommands can be created with
-respective macros: :c:macro:`SHELL_CMD_REGISTER` and :c:macro:`SHELL_CMD`.
-In addition, you can define options for commands or subcommands using the
-macro :c:macro:`SHELL_OPT`. By default, each and every command or subcommand
-has these two options implemented: ``-h`` and ``--help``.
+Every user-defined command or subcommand can have its own help description.
+The help for commands and subcommands can be created with respective macros:
+:c:macro:`SHELL_CMD_REGISTER`, :c:macro:`SHELL_CMD_ARG_REGISTER`,
```

Comment:
It can be updated in separate PR.
If you will register a command with `SHELL_CMD_ARG_REGISTER` shell will check if command has be called with required number of arguments. You will not need to do it in command handler

Reply:
ARG variant takes the number of mandatory and optional arguments so they can be validated with use of precheck.

Reply:
I've updated macro description